OmniKit/PumpManager/OmnipodPumpManager.swift

```diff
@@ -1580,30 +1580,46 @@ extension OmnipodPumpManager: PumpManager {
                 return
             }
 
-            let status: StatusResponse
+            // resuming a normal basal is denoted by a 0 duration temp basal which simply cancels any existing temp basal
+            let resumingNormalBasal = duration < .ulpOfOne
 
-            let result = session.cancelDelivery(deliveryType: .tempBasal)
-            switch result {
-            case .certainFailure(let error):
-                completion(.communication(error))
-                return
-            case .uncertainFailure(let error):
-                // TODO: Return PumpManagerError.uncertainDelivery and implement recovery
-                completion(.communication(error))
-                return
-            case .success(let cancelTempStatus, _):
-                status = cancelTempStatus
-            }
+            // Skip the Cancel TB comms optimization if the last message had any
+            // comms issues or if the last delivery status hasn't been verified OK
+            let skipCancelTBCommsOptimization = self.state.podState?.lastCommsOK == false ||
```

Comment:
Maybe rename `skipCancelTBCommsOptimization` to `needStatusCheck`?

Reply:
I changed this variable to uncertainDeliveryStatus as it more appropriate. We don't need to a status check in these cases, we just need to do the cancel temp basal command because we can't be certain of the current delivery status. The skipCancelTBCommsOptimization variable name was originally chosen to try make it easier for someone who wanted to disable this optimization.